Code/GraphMol/CIPLabeler/catch_tests.cpp

```diff
@@ -936,4 +942,140 @@ TEST_CASE("atropisomers", "[basic]") {
+
+    if (mol == NULL) {
+      mol = MolBlockToMol(molBlock, false, false);
+      mol->updatePropertyCache(false);
+    }
+
+    std::vector<std::string> atomsArray;
+    std::vector<std::string> bondsArray;
+
+    RDKit::CIPLabeler::assignCIPLabels(*mol, maxRecursiveIterations);
+    for (auto atom : mol->atoms()) {
+      if (atom->hasProp(common_properties::_CIPCode)) {
+        std::string thisVal =
+            atom->getProp<std::string>(common_properties::_CIPCode);
+        atomsArray.push_back(thisVal.c_str());
+      } else {
+        atomsArray.push_back("");  // push null value
+      }
+    }
+
+    for (auto bond : mol->bonds()) {
+      if (bond->hasProp(common_properties::_CIPCode)) {
+        std::string thisVal =
+            bond->getProp<std::string>(common_properties::_CIPCode);
+        bondsArray.push_back(thisVal.c_str());
+      } else {
+        bondsArray.push_back("");
+      }
+    }
+
+    result = boost::algorithm::join(atomsArray, ", ") + ":" +
+             boost::algorithm::join(bondsArray, ", ");
+    return result;
+  } catch (const CIPLabeler::MaxIterationsExceeded &e) {
+    delete mol;
+    mol = nullptr;
+    return "";
+  } catch (const std::exception &e) {
+    delete mol;
+    mol = nullptr;
+    return e.what();
+  }
+}
+
+TEST_CASE("CipLabelAtropsOnRing", "[basic]") {
+  SECTION("atropisomers1") {
+    std::string molBlock = R"(
+  Mrv2304 04172300142D
+
+  0  0  0     0  0            999 V3000
+M  V30 BEGIN CTAB
+M  V30 COUNTS 24 26 0 0 1
+M  V30 BEGIN ATOM
+M  V30 1 C -0.0002 1.5403 0 0
+M  V30 2 N -0.0002 3.0805 0 0 CFG=1
+M  V30 3 C -1.334 3.8508 0 0
+M  V30 4 C -2.6678 3.0807 0 0
+M  V30 5 C -1.334 5.391 0 0
+M  V30 6 C 1.3338 5.391 0 0
+M  V30 7 C 1.3338 3.8508 0 0
+M  V30 8 Br 2.6676 3.0807 0 0
+M  V30 9 C -1.3338 0.7702 0 0
+M  V30 10 C -2.6678 1.5403 0 0
+M  V30 11 C -1.3338 -0.7702 0 0
+M  V30 12 C -2.6678 -1.5401 0 0
+M  V30 13 C -0.0002 -1.5403 0 0
+M  V30 14 N -0.0002 -3.0805 0 0 CFG=1
+M  V30 15 C 1.3338 -3.8508 0 0
+M  V30 16 C 2.6676 -3.0805 0 0
+M  V30 17 C 1.3338 -5.391 0 0
+M  V30 18 C -1.334 -5.391 0 0
+M  V30 19 C -1.334 -3.8508 0 0
+M  V30 20 Br -2.6678 -3.0805 0 0
+M  V30 21 C 1.3338 -0.7702 0 0
+M  V30 22 C 2.6678 -1.5403 0 0
+M  V30 23 C 1.3338 0.7702 0 0
+M  V30 24 C 2.6678 1.5404 0 0
+M  V30 END ATOM
+M  V30 BEGIN BOND
+M  V30 1 1 1 2
+M  V30 2 1 2 3
+M  V30 3 1 3 4
+M  V30 4 2 3 5
+M  V30 5 1 5 6
+M  V30 6 2 6 7
+M  V30 7 1 2 7 CFG=1
+M  V30 8 1 7 8
+M  V30 9 2 1 9
+M  V30 10 1 9 10
+M  V30 11 1 9 11
+M  V30 12 1 11 12
+M  V30 13 2 11 13
+M  V30 14 1 13 14
+M  V30 15 1 14 15 CFG=1
+M  V30 16 1 15 16
+M  V30 17 2 15 17
+M  V30 18 1 17 18
+M  V30 19 2 18 19
+M  V30 20 1 14 19
+M  V30 21 1 19 20
+M  V30 22 1 13 21
+M  V30 23 1 21 22
+M  V30 24 2 21 23
+M  V30 25 1 1 23
+M  V30 26 1 23 24
+M  V30 END BOND
+M  V30 END CTAB
+M  END
+ )";
+
+    UseLegacyStereoPerceptionFixture useLegacy(false);
+    auto found = cipLabels(molBlock, 5000000);
```

Comment:
I think your goal in this test is to make sure that the stereo property hasn't been removed from the two atropisomeric bonds.

Rather than going through the trouble of calculating CIPs and then testing for them in this complicated manner, it would be better to directly check whether or not the stereo information is still present by checking the return value of `getStereo()` on the two bonds you care about.

Then you can also move this test out of the CIP labeller and put it in `catch_chirality.cpp`

Reply:
The problem was initially found because CIP labeling did not work correctly.  But the issue, as you pointed out, is that the atropisomer designation was being removed.   I moved the test as suggested.